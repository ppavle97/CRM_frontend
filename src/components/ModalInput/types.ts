
export interface ModalInputProps {
  state: any;
  setState: any;
  label: string;
  type: string;
}
