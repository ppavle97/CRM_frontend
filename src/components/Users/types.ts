import { User } from "../../store/features/user/userTypes";

export interface StyledUserItemProps {
  active?: boolean;
}

export interface UsersProps {
  users?: User[]
}
