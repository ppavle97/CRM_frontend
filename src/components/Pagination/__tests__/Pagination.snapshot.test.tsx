import { render } from "@testing-library/react";
import { Pagination } from "..";
import { PaginationProps } from "../types";

describe("Pagination component", () => {
  const defaultProps: PaginationProps = {
    currentPage: 1,
    totalPages: 5,
    onPageChange: jest.fn(),
  };

  it("matches snapshot with default props", () => {
    const { asFragment } = render(<Pagination {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot with updated props", () => {
    const { asFragment, rerender } = render(<Pagination {...defaultProps} />);

    rerender(<Pagination {...defaultProps} currentPage={3} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
