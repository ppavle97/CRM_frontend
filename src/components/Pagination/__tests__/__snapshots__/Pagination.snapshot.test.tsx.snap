// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="css-1yp1utw"
  >
    <button
      class="css-e9qjh2"
      disabled=""
    >
      <img
        alt="left-arrow"
      />
    </button>
    <button
      class="css-16gpifp"
    >
      1
    </button>
    <button
      class="css-e9qjh2"
    >
      2
    </button>
    <span
      class="css-1mhloja"
    >
      ...
    </span>
    <button
      class="css-e9qjh2"
    >
      5
    </button>
    <button
      class="css-e9qjh2"
    >
      <img
        alt="right-arrow"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`Pagination component matches snapshot with updated props 1`] = `
<DocumentFragment>
  <div
    class="css-1yp1utw"
  >
    <button
      class="css-e9qjh2"
    >
      <img
        alt="left-arrow"
      />
    </button>
    <button
      class="css-e9qjh2"
    >
      1
    </button>
    <button
      class="css-e9qjh2"
    >
      2
    </button>
    <button
      class="css-16gpifp"
    >
      3
    </button>
    <button
      class="css-e9qjh2"
    >
      4
    </button>
    <button
      class="css-e9qjh2"
    >
      5
    </button>
    <button
      class="css-e9qjh2"
    >
      <img
        alt="right-arrow"
      />
    </button>
  </div>
</DocumentFragment>
`;
